### ISSUE TITLE
As iBabs-application I want to get user-auth filtered results from the API endpoints

### ISSUE THREAD

**Issue by sergei-maertens** [GemeenteUtrecht/ZGW#446]
...so that end-users in iBabs only see data that they're authorized to see.

The strategy is to implement a proxy/middleware API in the ZAC, where permissions are currently defined.

iBabs will make API calls with a custom header containing the User-ID (username or perhaps e-mail) to this proxy, and the (ZAC) proxy will make the matching API call to the ZRC/DRC/... It's the ZAC's job to filter out results and reply only with the relevant results to the caller.

This achieves:

1. Single location for permission definitions for users with respect to Zaaktype/confidentiality
2. iBabs has the same API spec as the "API's voor zaakgericht werken" standard, nothing changes

Possibly this will be moved to its own microservice eventually, but let's keep it close now for performance reasons.

**Comment by MichelleDautzenberg** [comment]
Begrijp ik goed dat dit een generiek component wordt dat door verschillende taakapplicaties gebruikt kan worden?

**Comment by sergei-maertens** [comment]
Ja, in principe wel. In eerste instantie bouwen we die wel gewoon in de ZAC in, maar dat kan later wel losgetrokken worden en dan zou de ZAC hier ook gebruik van kunnen maken.
